File: LeetCode/2110. Number of Smooth Descent Periods of a Stock.cpp
class Solution {
public:
    long long getDescentPeriods(vector<int>& prices) {
        long long len=1, ans=0, prev=-1;

        for(auto p: prices) {
            if(prev - p == 1) {
                len++;
                ans += len;
            } else {
                len = 1;
                ans++;
            }

            prev = p;
        }

        return ans;
    }
};

File: LeetCode/2110. Number of Smooth Descent Periods of a Stock.go
func getDescentPeriods(prices []int) int64 {

    var len int64 = 1
    var ans int64 = 0
    prev := -1

    for _, v := range prices {
        if prev - v == 1 {
            len++
            ans += len
        } else {
            ans++;
            len = 1
        }

        prev = v
    }

    return ans
}